internal/services/access_flow.go

```diff
@@ -0,0 +1,476 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"github.com/apono-io/apono-sdk-go"
+	"github.com/apono-io/terraform-provider-apono/internal/models"
+	"github.com/apono-io/terraform-provider-apono/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"golang.org/x/exp/slices"
+	"math/big"
+	"strings"
+)
+
+func ConvertToAccessFlowModel(ctx context.Context, aponoClient *apono.APIClient, accessFlow *apono.AccessFlowV1) (*models.AccessFlowModel, diag.Diagnostics) {
+	data := models.AccessFlowModel{}
+	data.ID = types.StringValue(accessFlow.GetId())
+	data.Name = types.StringValue(accessFlow.GetName())
+	data.Active = types.BoolValue(accessFlow.GetActive())
+	data.RevokeAfterInSec = types.NumberValue(big.NewFloat(float64(accessFlow.GetRevokeAfterInSec())))
+
+	trigger := accessFlow.GetTrigger()
+	dataTrigger := models.Trigger{
+		Type: types.StringValue(trigger.GetType()),
+	}
+
+	timeframe, ok := trigger.GetTimeframeOk()
+	if ok && timeframe != nil {
+		var timeframeDaysAsStrings []types.String
+		for _, day := range timeframe.GetDaysInWeek() {
+			timeframeDaysAsStrings = append(timeframeDaysAsStrings, types.StringValue(string(day)))
+		}
+		daysInWeek, diagnostics := types.SetValueFrom(ctx, types.StringType, timeframeDaysAsStrings)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		startTimeString := utils.SecondsToDayTimeFormat(int(timeframe.GetStartOfDayTimeInSeconds()))
+		endTimeString := utils.SecondsToDayTimeFormat(int(timeframe.GetEndOfDayTimeInSeconds()))
+
+		dataTrigger.Timeframe = &models.Timeframe{
+			StartTime:  types.StringValue(startTimeString),
+			EndTime:    types.StringValue(endTimeString),
+			DaysInWeek: daysInWeek,
+			TimeZone:   types.StringValue(timeframe.GetTimeZone()),
+		}
+
+	}
+	data.Trigger = &dataTrigger
+
+	availableIdentities, _, err := aponoClient.IdentitiesApi.ListIdentities(ctx).Execute()
+	if err != nil {
+		return nil, utils.GetDiagnosticsForApiError(err, "list", "identities", "")
+	}
+
+	var dataGrantees []models.Identity
+	for _, grantee := range accessFlow.GetGrantees() {
+		identity, diagnostics := convertToIdentityModel(ctx, grantee.Id, strings.ToLower(grantee.Type), aponoClient, availableIdentities.Data)
+		if len(diagnostics) > 0 {
+			return nil, diagnostics
+		}
+
+		dataGrantees = append(dataGrantees, *identity)
+	}
+	setGrantees, diags := types.SetValueFrom(ctx, types.ObjectType{AttrTypes: models.IdentityObject}, getUniqueListOfIdentities(dataGrantees))
+	if len(diags) > 0 {
+		return nil, diags
+	}
+	data.Grantees = setGrantees
```

Comment:
I dont get why you need this conversion and why there is Identity and IdentityObject

Reply:
add comment about the conversion for types